### PR TITLE
Lookup lower distances if there's space

### PR DESCRIPTION
**Pull request by jtraglia**
## PR Description

Alternative version of #164 based on [this comment](https://github.com/ConsenSys/discovery/pull/164#issuecomment-1297698736). We want to prioritize higher distances, but...

> It would be reasonable to add lower distances if we've reached MAX_DISTANCE though.

This PR does exactly that.

### CODE REVIEW

File: src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java [hunk 1]
```diff
@@ -96,14 +100,38 @@ private CompletableFuture<Collection<NodeRecord>> performSearchForNewPeers() {
...
-        distance++) {
-      distances.add(distance);
+    distances.add(targetDistance);
+
+    // We want to prioritize higher distances in the request over lower ones
+    // because lower distances are less likely to contain nodes.
+    for (int offset = 1; targetDistance + offset <= KBuckets.MAXIMUM_BUCKET; offset++) {
+      if (distances.size() >= LOOKUP_REQUEST_LIMIT) {
+        break;
+      }
```
Comment (ajsutton) [inline src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java:118]: nit: I think I'd simplify this to:
```suggestion
    for (int offset = 1; targetDistance + offset <= KBuckets.MAXIMUM_BUCKET && distances.size() < LOOKUP_REQUEST_LIMIT; offset++) {
```

File: src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java [hunk 2]
```diff
@@ -96,14 +100,38 @@ private CompletableFuture<Collection<NodeRecord>> performSearchForNewPeers() {
...
+      distances.add(targetDistance + offset);
+    }
+
+    // If there's any space left, add lower distances.
+    for (int offset = 1; targetDistance - offset >= KBuckets.MINIMUM_BUCKET; offset++) {
+      if (distances.size() >= LOOKUP_REQUEST_LIMIT) {
+        break;
+      }
```
Comment (ajsutton) [inline src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java:126]: Similar simplification here.

**Review by ajsutton — COMMENTED** [review]
LGTM just a couple of minor simplifications I'd suggest.

File: src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java (hunk 1 above)
Comment (jtraglia) [reply src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java]: Good call. I've updated those.

**Review by ajsutton — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**